src/SAML2/XML/samlp/AbstractMessage.php

```diff
@@ -431,8 +420,10 @@ public function setRelayState(string $relayState = null): void
      *
      * @return \DOMElement The root element of the DOM tree
      */
-    public function toXML(): DOMElement
+    public function toXML(?DOMElement $parent = null): DOMElement
     {
+        Assert::null($parent);
```

Comment:
I think there can be situations where a message is not the actual root of the XML document (SOAP? PAOS?), so we should actually use `$parent` here.